src/sentry/ingest/transaction_clusterer/rules.py

```diff
@@ -89,7 +95,25 @@ def write(self, project: Project, rules: RuleSet) -> None:
     def merge(self, project: Project) -> None:
         """Read rules from all stores, merge and write them back so they all are up-to-date."""
         merged_rules = self.read(project)
-        self.write(project, merged_rules)
+        trimmed_rules = self._trim_rules(merged_rules)
+        self.write(project, trimmed_rules)
+
+    def _trim_rules(self, rules: RuleSet) -> RuleSet:
```

Comment:
Instead of injecting the trimming logic in `merge`, I thought about creating another component and passing it to `CompositeRuleStore`. I decided not to do that because I believe it's overengineering this solution for now. We can explore a similar solution, or revisit this in the future when we have injected more "limiting" logic. I didn't want to modify the "regular" rule stores either (e.g. `RedisRuleStore`) for being more practical to reuse this code, and to leave them with the single responsibility of storing.